commands/project_sync.go

```diff
@@ -86,35 +86,228 @@ func (cmd *ProjectSync) Commands() []cli.Command {
+		cmd.out.Error("Unison container (%s) is not running", volumeName)
+		return cmd.Failure(err.Error(), "SYNC-CHECK-FAILED", 13)
+	}
+	cmd.out.Info("Unison container found: %s", volumeName)
+	cmd.out.Spin("Check unison container process is listening...")
+	if _, err := cmd.WaitForUnisonContainer(volumeName, ctx.Int("initial-sync-timeout")); err != nil {
+		cmd.out.Error("Unison process not listening")
+		return cmd.Failure(err.Error(), "SYNC-CHECK-FAILED", 13)
+	}
+	cmd.out.Info("Unison process is listening")
+
+	// Determine if sync progress can be tracked.
+	//cmd.out.Spin("Syncing a test file...")
+	cmd.out.Info("Preparing live file sync test")
+	var logFile = cmd.LogFileName(volumeName)
+	if err := cmd.WaitForSyncInit(logFile, workingDir, ctx.Int("initial-sync-timeout"), ctx.Int("initial-sync-wait")); err != nil {
+		return cmd.Failure(err.Error(), "UNISON-SYNC-FAILED", 13)
+	}
+
+	// Sidestepping the notification so rig sync:check can be run as a background process.
+	cmd.out.Info("Sync check completed successfully")
+	return nil
+}
+
+// RunPurge cleans out the project sync state.
+func (cmd *ProjectSync) RunPurge(ctx *cli.Context) error {
+	if util.IsLinux() {
+		return cmd.Success("No Unison process to clean up.")
+	}
+
+	volumeName, workingDir, err := cmd.initializeSettings(ctx.String("dir"))
+	if err != nil {
+		return cmd.Failure(err.Error(), "SYNC-PATH-ERROR", 12)
+	}
+
+	cmd.out.Spin("Checking for unison container...")
+	if running := util.ContainerRunning(volumeName); running {
+		cmd.out.Spin(fmt.Sprintf("Stopping Unison container (%s)", volumeName))
+		if err := util.Command("docker", "container", "stop", volumeName).Run(); err != nil {
+			cmd.out.Warn("Could not stop unison container (%s): Maybe it's already stopped?", volumeName)
+		} else {
+			cmd.out.Info("Stopped unison container (%s)", volumeName)
+		}
+	} else {
+		cmd.out.Info("No running unison container.")
+	}
+
+	logFile := cmd.LogFileName(volumeName)
+	cmd.out.Spin(fmt.Sprintf("Removing unison log file: %s", logFile))
+	if util.FileExists(logFile, workingDir) {
+		if err := util.RemoveFile(logFile, workingDir); err != nil {
+			cmd.out.Error("Could not remove unison log file: %s: %s", logFile, err.Error())
+		} else {
+			cmd.out.Info("Removed unison log file: %s", logFile)
+		}
+	} else {
+		cmd.out.Info("Log file does not exist")
+	}
+
+	// Remove sync fragment files.
+	cmd.out.Spin("Removing .unison directories")
+	if err := util.RemoveFileGlob("*.unison*", workingDir, cmd.out); err != nil {
+		cmd.out.Warning("Could not remove .unison directories")
+	} else {
+		cmd.out.Info("Removed all .unison directories")
+	}
+
+	cmd.out.Spin(fmt.Sprintf("Removing sync volume: %s", volumeName))
+	// @TODO capture the volume rm error text to display to user!
+	out, err := util.Command("docker", "volume", "rm", volumeName, "1>&2").Output()
+	if err != nil {
+		fmt.Println(string(out))
+		fmt.Println(err.Error())
+		return cmd.Failure(string(out), "SYNC-VOLUME-FAILURE", 13)
```

Comment:
I seem to always hit this condition. Here is a quick test script that always fails from me. It is set up to run from the rig project directory and use a locally built version of rig.

```
#!/bin/bash

set -e

DIR=purgetest
mkdir -p $DIR
./build/darwin/rig --verbose project sync:start --dir=$DIR
./build/darwin/rig --verbose project sync:stop --dir=$DIR
./build/darwin/rig --verbose project sync:purge --dir=$DIR
```

Reply:
I'll note the underlying command that the --verbose flag causes to spit out seems to run without error and the return code when it finishes is 0. Perhaps it has something to do with tie-ing the out/err file handles together

Reply:
Doing some experimenting it is the file handle tie-ing that is causing the issue. If I remove that part of the argument and remove the print of out in the case where the command succeeds it seems to run quietly which is what I thought the intention was.

Here is the diff that I tested.

```
diff --git a/commands/project_sync.go b/commands/project_sync.go
index ddfbee6..f27177b 100644
--- a/commands/project_sync.go
+++ b/commands/project_sync.go
@@ -278,13 +278,12 @@ func (cmd *ProjectSync) RunPurge(ctx *cli.Context) error {

        cmd.out.Spin(fmt.Sprintf("Removing sync volume: %s", volumeName))
        // @TODO capture the volume rm error text to display to user!
-       out, err := util.Command("docker", "volume", "rm", volumeName, "1>&2").Output()
+       out, err := util.Command("docker", "volume", "rm", volumeName).Output()
        if err != nil {
                fmt.Println(string(out))
                fmt.Println(err.Error())
                return cmd.Failure(string(out), "SYNC-VOLUME-FAILURE", 13)
        }
-       fmt.Println(string(out))

        return nil
 }
```